### PR TITLE
Migrate new RCS test to use the new REST test plugin

### PR DESCRIPTION
**Pull request by ywangd**
This PR migrates the new RCS REST test to use the new plugin. The inline clusters make it possible to add more test classes within the same package (will be added as follow-ups).

### CODE REVIEW

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java [hunk 1]
```diff
@@ -171,15 +171,25 @@ public String getTransportEndpoint() {
             return readPortsFile(portsFile).get(0);
         }
 
+        public String getRemoteClusterServerEndpoint() {
+            Path portsFile = workingDir.resolve("logs").resolve("remote_cluster.ports");
```
Comment (mark-vieira) [inline test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java:175]: Just for my education as I'm not familiar with this. So this remote cluster feature actually uses a unique transport separate from the normal transport address?

File: x-pack/plugin/security/qa/multi-cluster/build.gradle [hunk 2]
```diff
@@ -5,129 +5,11 @@
...
-
-  setting 'cluster.remote.my_remote_cluster.mode', 'proxy'
-  setting 'cluster.remote.my_remote_cluster.proxy_address', {
-    "\"${fulfillingCluster.get().getAllRemoteAccessPortURI().get(0)}\""
-  }
-  setting 'cluster.remote.my_remote_cluster.authorization', 'foobar'
-
-  user username: "test_user", password: "x-pack-test-password"
+dependencies {
+  javaRestTestImplementation project(':x-pack:plugin:core')
```
Comment (mark-vieira) [inline x-pack/plugin/security/qa/multi-cluster/build.gradle:12]: Why do we need the dependencies on these two x-pack modules? I don't see any references to them.

**Review by mark-vieira — APPROVED** [review]

File: x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java [hunk 3]
```diff
@@ -34,24 +44,76 @@
...
+        .configFile("transport.crt", Resource.fromClasspath("ssl/transport.crt"))
+        .configFile("transport-ca.crt", Resource.fromClasspath("ssl/transport-ca.crt"))
+        .configFile("remote-cluster.key", Resource.fromClasspath("ssl/remote_cluster.key"))
+        .configFile("remote-cluster.crt", Resource.fromClasspath("ssl/remote_cluster.crt"))
+        .configFile("remote-cluster-ca.crt", Resource.fromClasspath("ssl/remote-cluster-ca.crt"))
+        .user(USER, PASS.toString());
+
+    public static ElasticsearchCluster fulfillingCluster = ElasticsearchCluster.local()
+        .name("fulfilling-cluster")
+        .nodes(1)
```
Comment (mark-vieira) [inline x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java:77]: This is a bit unnecessary. The default behavior is a single-node cluster although it's fine to keep this if you want to be extra explicit.

**Review by mark-vieira — APPROVED** [review]

**Comment by mark-vieira** [comment]
Couple questions and comments but otherwise this looks great. Thanks for updating this test!

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java (hunk 1 above)
Comment (ywangd) [reply test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java:175]: Yes. We had some back and forth on the decision. But yes the new remote cluster feature has its own separate port. It still speaks the transport protocol.

File: x-pack/plugin/security/qa/multi-cluster/build.gradle (hunk 2 above)
Comment (ywangd) [reply x-pack/plugin/security/qa/multi-cluster/build.gradle:12]: Err they were copy/paste from ServiceAccountIT. Removed.

File: x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java (hunk 3 above)
Comment (ywangd) [reply x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java]: That's good to know. I removed the explicit config.

**Pull request merged**